checkov/terraform/graph_builder/local_graph.py

```diff
@@ -352,10 +362,14 @@ def _get_dest_module_path(self, curr_module_dir: str, dest_module_source: str, d
         return os.path.realpath(dest_module_path)
 
     def _find_vertex_index_relative_to_path(
-        self, block_type: BlockType, name: str, block_path: str, module_path: str, module_num: str
+        self, block_type: BlockType, name: str, block_path: str, module_path: str, module_num: str, relative_module_idx: Optional[int] = None
     ) -> int:
         relative_vertices = []
-        possible_vertices = self.vertices_by_module_dependency_by_name.get((module_path, module_num), {}).get(block_type, {}).get(name, [])
+        if relative_module_idx is not None:
+            module_dependency_by_name_key = next(k for k, v in self.vertices_by_module_dependency.items() if v.get(BlockType.MODULE, []).__contains__(relative_module_idx))
```

Comment:
Can we wrap a try..except block to catch possible `StopIteration` error?

Reply:
the `vertices_by_module_dependency` map is containing all the vertices in the graph mapped by modules so we shouldn't get this error